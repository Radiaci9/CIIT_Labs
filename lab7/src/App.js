import React from 'react'
import { Router, Route, Switch } from 'react-router-dom'
import { Provider } from 'react-redux'

import NavBar from './components/NavBar'
import RepositoriesPage from './containers/RepositoriesPage'
import Home from './containers/Home'
import AnotherPage from './containers/AnotherPage'

import store from './store'
import history from './history'

let navItems = [
  {
    label: 'Home Page',
    click: () => history.push('/')
  },
  {
    label: 'Page Repositories and Followers',
    click: () => history.push('/anotherPage')
  },
  {
    label: 'Best Repositories',
    click: () => history.push('/repositoriesPage')
  }
]
    
    export default class App extends React.Component {      
      render() {
        return (
          <Provider store={store}>
            <Router history={history}>
                <div>
                  <NavBar items = {navItems}/>
                  <Switch>
                      <Route exact string path='/' component={Home}/>
                      <Route exact string path='/anotherPage' component={AnotherPage}/>
                      <Route exact string path='/repositoriesPage' component={RepositoriesPage}/>
                  </Switch>
                </div>
            </Router>
        </Provider>
        )
      }
    }