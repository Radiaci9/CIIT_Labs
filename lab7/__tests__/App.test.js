import App from '../src/App'

describe('renders()', () => {
      test('renders without crashing', () => {
          expect(
            JSON.stringify(
              Object.assign({}, App, { _reactInternalInstance: 'censored' }),
            ),
          ).toMatchSnapshot()
      })
})