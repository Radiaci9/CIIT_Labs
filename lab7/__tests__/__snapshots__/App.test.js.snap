// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders() renders without crashing 1`] = `"{\\"_reactInternalInstance\\":\\"censored\\"}"`;
